File: build_management/bazel_primer/python_tutorial/stage1/main/hello_world.py
from datetime import datetime #import the datetime library

def get_greet(who):
    '''
    get_greet returns a concatenated string for our eventual greeting
    '''
    return "Hello " + str(who);
    
def print_localtime():
    '''
    print_localtime prints a locale string of the current date and
    time with format "Weekday(abbr.) Month(abbr.) dd hh:mm:ss YYYY"
    '''
    now = datetime.now()
    print(now.strftime("%a %b %d %H:%M:%S %Y"))	

if __name__ == "__main__":
    '''
    this function puts things together and calls our functions
    '''
    who = "world"
    print(get_greet(who))
    print_localtime()

File: build_management/bazel_primer/c_tutorial/stage1/main/hello_world.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

char* get_greet(char* who) {
  char *str = (char *) malloc(1024);
  strcpy(str, "Hello ");
  strcat(str, who);
  return str;
}

void print_localtime() {
  time_t t;
  time(&t);
  printf("\n%s", ctime(&t));
}

int main(int argc, char** argv) {
  char* who = "world";
  printf("%s", get_greet(who));
  print_localtime();
  return 0;
}
